src/attribute-filters.tsx

```diff
@@ -0,0 +1,376 @@
+    query.removePipelineStage({ operator: '|=' });
+  }
+
+  const severityPipelineStage = getSeverityFilterPipelineStage(filters);
+
+  if (severityPipelineStage) {
+    query
+      .removePipelineStage({}, { matchLabel: 'level' })
+      .addPipelineStage(getSeverityFilterPipelineStage(filters), {
+        placement: 'end',
+      });
+  }
+
+  if (filters?.severity === undefined || filters.severity.size === 0) {
+    query.removePipelineStage({}, { matchLabel: 'level' });
+  }
+
+  query.addSelectorMatcher(getMatchersFromFilters(filters));
+
+  attributes.forEach(({ id, label }) => {
+    if (label) {
+      const filterValue = filters?.[id];
+      if (filterValue === undefined || filterValue.size === 0) {
+        query.removeSelectorMatcher({ label });
+      }
+    }
+  });
+
+  return query.toString();
+};
+
+const removeQuotes = (value: string) => value.replace(/"/g, '');
+
+export const filtersFromQuery = ({
+  query,
+  attributes,
+}: {
+  query?: string;
+  attributes: AttributeList;
+}): Filters => {
+  const filters: Filters = {};
+  const logQLQuery = new LogQLQuery(query ?? '');
+
+  attributes.map(({ label, id }) => {
+    for (const selector of logQLQuery.streamSelector) {
+      if (label && label.length > 0 && selector.label === label) {
+        filters[id] = new Set(selector.value.split('|').map(removeQuotes));
+      }
+    }
+  });
+
+  for (const pipelineStage of logQLQuery.pipeline) {
+    if (
+      pipelineStage.operator === '|' &&
+      pipelineStage.labelsInFilter?.every(({ label }) => label === 'level') &&
+      !filters.severity
+    ) {
+      const severityValues: Array<Severity> = pipelineStage.labelsInFilter
+        .flatMap(({ value }) => value.split('|'))
+        .map(removeQuotes)
+        .map(severityFromString)
+        .filter(notUndefined);
+      filters.severity = new Set(severityValues);
+    } else if (pipelineStage.operator === '|=' && !filters.content) {
+      filters.content = new Set([removeQuotes(pipelineStage.value)]);
+    }
+  }
+
+  return filters;
+};
+
+export const getNamespaceMatcher = (
+  namespace?: string,
+): LabelMatcher | undefined => {
+  if (namespace === undefined) {
+    return undefined;
+  }
+
+  return {
+    label: 'kubernetes_namespace_name',
+    operator: '=',
+    value: `"${namespace}"`,
+  };
+};
+
+export const queryWithNamespace = ({
+  query,
+  namespace,
+}: {
+  query: string;
+  namespace?: string;
+}) => {
+  const logQLQuery = new LogQLQuery(query ?? '');
+
+  logQLQuery.addSelectorMatcher(getNamespaceMatcher(namespace));
+
+  return logQLQuery.toString();
+};
+
+export const getMatcherFromSet = (
+  label: string,
+  values: Set<string>,
+): LabelMatcher | undefined => {
+  const valuesArray = Array.from(values);
+  if (valuesArray.length === 0) {
+    return undefined;
+  }
+
+  if (valuesArray.length === 1) {
+    return {
+      label,
+      operator: '=',
+      value: `"${valuesArray[0]}"` ?? '',
```

Comment:
If `valuesArray[0]` is empty, this will be like `` `""` ?? ''``, so the `?? ''` will never be used.

Also, for empty strings here you would get `label=""`, but for the length > 1 case below you filter out empty values. Shouldn't the behavior be similar?